### PR TITLE
[feat] Cargo Service

### PR DESCRIPTION
**Pull request by Ghazaleh7979**

### CODE REVIEW

File: Examples/CargoFactory/ShipTransferFactory.cs [hunk 1]
```diff
@@ -0,0 +1,15 @@
+﻿namespace Examples.Cargo
+{
+    public class ShipTransferFactory : ITransferFactory
+    {
+        public ITransfer Send(string origin, string destination)
+        {
+            return new ShipTransfer(origin, destination);
+        }
+
+        public ITransfer Send()
```
Comment (mohammadKarimi) [inline Examples/CargoFactory/ShipTransferFactory.cs:10]: why you define two methods to force classes to implement and throw exception

File: Examples/CargoFactory/ITransferFactory.cs [hunk 2]
```diff
@@ -0,0 +1,8 @@
+﻿namespace Examples.Cargo
+{
+    public interface ITransferFactory
+    {
+        ITransfer Send();
+        ITransfer Send(string origin, string destination);
```
Comment (mohammadKarimi) [inline Examples/CargoFactory/ITransferFactory.cs:6]: it is not good idea.

File: Examples/CargoFactory/AIRTransferFactory.cs [hunk 3]
```diff
@@ -0,0 +1,15 @@
+﻿namespace Examples.Cargo
+{
+    public class AIRTransferFactory : ITransferFactory
```
Comment (mohammadKarimi) [inline Examples/CargoFactory/AIRTransferFactory.cs:3]: you have air factory, and it's responsibility is create and manage the instance of Air. but you did it in Air class.

File: Examples/Cargo/AIRTransfer.cs [hunk 4]
```diff
@@ -0,0 +1,11 @@
+﻿namespace Examples.Cargo
```
Comment (mohammadKarimi) [inline Examples/Cargo/AIRTransfer.cs:1]: use file scope namespace.

File: Examples/Cargo/AIRTransfer.cs [hunk 5]
```diff
@@ -0,0 +1,11 @@
...
+{
+    public class AIRTransfer : ITransfer
+    {
+        private static AIRTransfer? _instance;
+        public void Send()
+        {
+            _instance = _instance ?? new AIRTransfer();
```
Comment (mohammadKarimi) [inline Examples/Cargo/AIRTransfer.cs:8]: it is not thread safe.
move this responsibilty to factory.

Comment (mohammadKarimi) [inline Examples/Cargo/AIRTransfer.cs:8]: why creating instance here inside the send method

**Review by mohammadKarimi — CHANGES_REQUESTED** [review]

**Comment by Ghazaleh7979** [comment]
I fixed your mentions

**Pull request closed without merging**